session/database.rst

```diff
@@ -0,0 +1,600 @@
+
+``db_id_col`` (default ``sess_id``):
+    The name of the column where to store the session ID (column type: ``VARCHAR(128)``);
+
+``db_data_col`` (default ``sess_data``):
+    The name of the column where to store the session data (column type: ``BLOB``);
+
+``db_time_col`` (default ``sess_time``):
+    The name of the column where to store the session creation timestamp (column type: ``INTEGER``);
+
+``db_lifetime_col`` (default ``sess_lifetime``):
+    The name of the column where to store the session lifetime (column type: ``INTEGER``).
+
+Preparing the Database to Store Sessions
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Before storing sessions in the database, you must create the table that stores
+the information. The session handler provides a method called
+:method:`Symfony\\Component\\HttpFoundation\\Session\\Storage\\Handler::createTable`
+to set up this table for you according to the database engine used::
+
+    try {
+        $sessionHandlerService->createTable();
+    } catch (\PDOException $exception) {
+        // the table could not be created for some reason
+    }
+
+If you prefer to set up the table yourself, it's recommended to generate an
+empty database migration with the following command:
+
+.. code-block:: terminal
+
+    $ php bin/console doctrine:migrations:generate
+
+Then, find the appropriate SQL for your database below, add it to the migration
+file and run the migration with the following command:
+
+.. code-block:: terminal
+
+    $ php bin/console doctrine:migrations:migrate
+
+MySQL
+.....
+
+.. code-block:: sql
+
+    CREATE TABLE `sessions` (
+        `sess_id` VARCHAR(128) NOT NULL PRIMARY KEY,
+        `sess_data` BLOB NOT NULL,
+        `sess_time` INTEGER UNSIGNED NOT NULL,
+        `sess_lifetime` MEDIUMINT NOT NULL
+    ) COLLATE utf8mb4_bin, ENGINE = InnoDB;
+
+.. note::
+
+    A ``BLOB`` column type can only store up to 64 kb. If the data stored in
+    a user's session exceeds this, an exception may be thrown or their session
+    will be silently reset. Consider using a ``MEDIUMBLOB`` if you need more
+    space.
+
+PostgreSQL
+..........
+
+.. code-block:: sql
+
+    CREATE TABLE sessions (
+        sess_id VARCHAR(128) NOT NULL PRIMARY KEY,
+        sess_data BYTEA NOT NULL,
+        sess_time INTEGER NOT NULL,
+        sess_lifetime INTEGER NOT NULL
+    );
+
+Microsoft SQL Server
+....................
+
+.. code-block:: sql
+
+    CREATE TABLE [dbo].[sessions](
+        [sess_id] [nvarchar](255) NOT NULL,
+        [sess_data] [ntext] NOT NULL,
+        [sess_time] [int] NOT NULL,
+        [sess_lifetime] [int] NOT NULL,
+        PRIMARY KEY CLUSTERED(
+            [sess_id] ASC
+        ) WITH (
+            PAD_INDEX  = OFF,
+            STATISTICS_NORECOMPUTE  = OFF,
+            IGNORE_DUP_KEY = OFF,
+            ALLOW_ROW_LOCKS  = ON,
+            ALLOW_PAGE_LOCKS  = ON
+        ) ON [PRIMARY]
+    ) ON [PRIMARY] TEXTIMAGE_ON [PRIMARY]
+
+Store Sessions in a NoSQL Database (MongoDB)
```

Comment:
why saying `in a NoSQL Database` when this section is only about MongoDB (while this is not the only NoSQL database) ?

Reply:
For Google and SEO reasons. It also makes the headline future-proof in case we add more nosql databases. Thanks!